client/layout/index.jsx

```diff
@@ -282,7 +283,11 @@ class Layout extends Component {
 				<HtmlIsIframeClassname />
 				<DocumentHead />
 				<QuerySites primaryAndRecent={ ! config.isEnabled( 'jetpack-cloud' ) } />
-				{ this.props.shouldQueryAllSites && <QuerySites allSites /> }
+				{ isJetpackCloud() ? (
+					<FetchSites />
+				) : (
+					this.props.shouldQueryAllSites && <QuerySites allSites />
+				) }
```

Comment:
This can be problematic. How can you guarantee that we'll use only the query on Jetpack Cloud and we'll use only the Redux state on Calypso blue? It's not a good idea IMHO.